openmp/libomptarget/plugins-nextgen/amdgpu/src/rtl.cpp

```diff
@@ -2728,6 +2749,30 @@ struct AMDGPUDeviceTy : public GenericDeviceTy, AMDGenericDeviceTy {
     return Err;
   }
 
+  /// Detect if current architecture is an APU.
+  Error checkIfAPU() {
+    std::string StrGfxName(ComputeUnitKind);
+    std::transform(std::begin(StrGfxName), std::end(StrGfxName),
+                   std::begin(StrGfxName),
+                   [](char c) { return std::tolower(c); });
+    if (StrGfxName == "gfx940") {
```

Comment:
This is pretty gross, I really wish this was exposed in a non-string way if it's not. If there's no way to make this any better just use a `StringSwitch`.